tutorials/basic-topics/linear-search.md

```diff
@@ -0,0 +1,149 @@
+---
+title: 'Linear search'
+description: 'Linear search is a searching technique that works in a linear/sequential fashion. It traverses whole of the data structure and returns the position(s) where the value is found. If not found then it simple returns nothing.'
+hide_table_of_contents: true
+---
+
+<TutorialAuthors names="@siddoinghisjob"/>
+
+## Overview
+
+Lets say we have a linear data structure - array, linked list - and we need to search for a certain element. To search there are mainly two searching techniques, one being linear search. In linear search we traverse the whole array and then while traversing we check for the particular item. If there's a match then we return that position( we can return multiple positions too).
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@siddoinghisjob"/>
+
+```cpp
+#include<iostream>
+using namespace std;
+
+int main(){
+  // Array to keep the values
+  int arr[5] = {1,2,3,4,5};
+  // Item to be searched 
+  int item = 3; 
+  for(int i = 0; i < 5; i++){
+    // checking if item is arr[i]
+    if(arr[i] == item){  
+      cout << "Found at " << i << endl;
+    }
+  }
+} 
+```
+
+</TabItem>
+</Tabs>
+
+## Example problems
+
+**Leetcode #1295 : Find Numbers with Even Number of Digits**
+
+Here we are given an array and we are required to find out the number of digits that are even.
+```
+array[] = {1,2,3,4,5,6}
+Answer = 3
+Explaination = We have 2, 4, 6 which are even numbers and therfore the output is 3.
+```
+
+***Solution -***
+
+We can think this problem as a linear search problem, where we are supposed to search for all the `even` numbers. Thats' it. Now to do so we will run a for loop
+and traverse every element and check it. If found then we will simply increase the count of even digits by 1.
+
+<Tabs>
+<TabItem value="cpp" label="C++">
+<SolutionAuthor name="@siddoinghisjob"/>
+
+```cpp
+#include<iostream>
+using namespace std;
+
+int main(){
+  // Array containing values
+  int arr[5] = {1,2,3,4,5}; 
+  // Initially the answer is zero
+  int answer = 0; 
+  for(int i = 0; i < 5; i++){
+     // check for division by 2 i.e., whether an even number
+    if(arr[i]%2 == 0) 
+      answer++;
+  }
+  cout << "Number of even digits are :- " << answer << endl;
+} 
+```
```

Comment:
This should be LC solution of problem 1295.

Reply:
The solution to this problem is not there in the repo, so I had to explain it here (this way).